### PR TITLE
Add version 2 info in statistics

### PR DESCRIPTION
**Pull request by ran-huang**
Signed-off-by: Ran <huangran@pingcap.com>

### What is changed, added or deleted? (Required)

Add descption about the analyze's changes based on the variable `tidb_analyze_version`

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/5164
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: statistics.md [hunk 1]
```diff
@@ -6,7 +6,25 @@ aliases: ['/docs/dev/statistics/','/docs/dev/reference/performance/statistics/']
...
+
+| Information | Version 1 | Version 2|
+| --- | --- | ---|
+| The total number of rows in the table | √ | √ |
+| Column Count-Min Sketch | √ | × |
+| Count-Min Sketch | √ | × |
+| Column Top-N | √ | √ (Maintenance methods and precision are improved) |
+| Index Top-N | √ (Insufficient maintenance precision might cause inaccuracy | √ (Maintenance methods and precision are improved) |
+| Column histogram | √ | √ (The histogram does not include Top-N values) |
+| Index histogram | √ | √ (The histogram buckets records the number different values in each bucket, and the histogram does not include Top-N values) |
```
Comment (TomShawn) [inline statistics.md:19]:
```suggestion
| Index histogram | √ | √ (The histogram buckets record the number of different values in each bucket, and the histogram does not include Top-N values.) |
```

File: statistics.md [hunk 2]
```diff
@@ -6,7 +6,25 @@ aliases: ['/docs/dev/statistics/','/docs/dev/reference/performance/statistics/']
...
+TiDB uses statistics to decide [which index to choose](/choose-index.md). The `tidb_analyze_version` variable controls the statistics collected by TiDB. Currently two versions are supported: `tidb_analyze_version = 1` (by default) and `tidb_analyze_version = 2`. The two versions are listed as follows:
+
+| Information | Version 1 | Version 2|
+| --- | --- | ---|
+| The total number of rows in the table | √ | √ |
+| Column Count-Min Sketch | √ | × |
+| Count-Min Sketch | √ | × |
+| Column Top-N | √ | √ (Maintenance methods and precision are improved) |
+| Index Top-N | √ (Insufficient maintenance precision might cause inaccuracy | √ (Maintenance methods and precision are improved) |
+| Column histogram | √ | √ (The histogram does not include Top-N values) |
```
Comment (TomShawn) [inline statistics.md:18]:
```suggestion
| Column histogram | √ | √ (The histogram does not include Top-N values.) |
```

File: statistics.md [hunk 3]
```diff
@@ -6,7 +6,25 @@ aliases: ['/docs/dev/statistics/','/docs/dev/reference/performance/statistics/']
...
+| Index histogram | √ | √ (The histogram buckets records the number different values in each bucket, and the histogram does not include Top-N values) |
+| The number of `NULL`s in the column | √ | √ |
+| The number of `NULL`s in the index | √ | √ |
+| The average length of columns | √ | √ |
+| The average length of indexes | √ | √ |
+
+Compared to Version 1, Version 2 statistics avoids the possible inaccuracy caused by hash collision in a large amount of data. It also incraease the estimate precision in most scenarios.
```
Comment (TomShawn) [inline statistics.md:25]:
```suggestion
Compared to Version 1, Version 2 statistics avoids the potential inaccuracy caused by hash collision when the data volume is huge. It also increases the estimate precision in most scenarios.
```

File: statistics.md [hunk 4]
```diff
@@ -248,6 +270,7 @@ Currently, the `SHOW STATS_HISTOGRAMS` statement returns the following 8 columns
 | `column_name` | The column name (when `is_index` is `0`) or the index name (when `is_index` is `1`) |
 | `is_index` | Whether it is an index column or not |
 | `update_time` | The time of the update |
+| `version` | The value of `tidb_analyze_version` in the corresponding `analyze` statement |
```
Comment (TomShawn) [inline statistics.md:273]:
```suggestion
| `version` | The value of `tidb_analyze_version` in the corresponding `ANALYZE` statement |
```

File: statistics.md [hunk 5]
```diff
@@ -284,6 +307,31 @@ Currently, the `SHOW STATS_BUCKETS` statement returns the following 10 columns:
 | `repeats` | The occurrence number of the maximum value |
 | `lower_bound` | The minimum value |
 | `upper_bound` | The maximum value |
+| `ndv` | The number of different values in the bucket. When `tidb_analyze_version` = 1, `ndv` is always 0, which has no actual meaning. |
```
Comment (TomShawn) [inline statistics.md:310]:
```suggestion
| `ndv` | The number of different values in the bucket. When `tidb_analyze_version` = `1`, `ndv` is always `0`, which has no actual meaning. |
```

**Review by TomShawn — APPROVED** [review]
Rest LGTM

**Review by winoros — COMMENTED** [review]
lgtm

**Pull request merged**